src/Controller/RegistrationController.php

```diff
@@ -1,20 +1,26 @@
 <?php
 
+declare(strict_types=1);
+
 namespace App\Controller;
 
 use App\Entity\User;
 use App\Form\RegistrationFormType;
-use Doctrine\ORM\EntityManagerInterface;
+use App\Repository\User\UserRepositoryInterface;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\PasswordHasher\Hasher\UserPasswordHasherInterface;
 use Symfony\Component\Routing\Annotation\Route;
 
-class RegistrationController extends AbstractController
+final class RegistrationController extends AbstractController
 {
+    public function __construct(
+        private readonly UserRepositoryInterface $userRepository
```

Comment:
```suggestion
        private readonly UserRepositoryInterface $userRepository,
```

Так же лучше UserPasswordHasherInterface тоже принимать как зависимость конструктора